Unit test for is_causal Better Transformers (#91900)

jcrousse opened the pull request:
Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/91900

Test Plan:
buck test  :test_transformers -- -r test_train_with_is_causal
buck test mode/opt :test_transformers -- -r test_is_causal_gpu
flake8 test_transformers.py

Differential Revision: D42453642

drisspg commented on test/test_transformers.py:
```diff
@@ -1381,16 +1383,76 @@ def test_train_with_is_causal(self, device):
+                        _ = mha_f(qkv_f, qkv_f, qkv_f, need_weights=False, is_causal=True)
+                        torch.cuda.synchronize()
+
+    @unittest.skipIf(not TEST_CUDA or not SM80OrLater or TEST_WITH_ROCM, "CUDA unavailable")
```
could you update to use the new https://github.com/pytorch/pytorch/blob/b7cad020b59ec2c9b9d62a1e6ca78c5b53f186a1/test/test_transformers.py#L36
if you are rebased recently enough.

jcrousse replied:
Done!

drisspg reviewed (approved):
Looks good besides the one comment

jcrousse committed:
Unit test for is_causal Better Transformers (#92102)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/92102

Pull Request resolved: https://github.com/pytorch/pytorch/pull/91900

Test Plan:
buck test  :test_transformers -- -r test_train_with_is_causal
buck test mode/opt :test_transformers -- -r test_is_causal_gpu
flake8 test_transformers.py

Reviewed By: drisspg

Differential Revision: D42453642

fbshipit-source-id: 6977634b1a69fe567276e50d2851301c7c2254e4

jcrousse commented:
@pytorchbot merge

drisspg commented:
@pytorchbot merge -f "the macos failures are unrelated"
